build-logic/src/main/java/qsl/internal/extension/QSLModuleExtension.java

```diff
@@ -0,0 +1,117 @@
+package qsl.internal.extension;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import javax.inject.Inject;
+
+import groovy.util.Node;
+import groovy.xml.QName;
+import org.gradle.api.Action;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.Dependency;
+import org.gradle.api.model.ObjectFactory;
+import org.gradle.api.plugins.JavaPlugin;
+import org.gradle.api.provider.Property;
+import org.gradle.api.publish.PublicationContainer;
+import org.gradle.api.publish.PublishingExtension;
+import org.gradle.api.publish.maven.MavenPublication;
+import org.gradle.api.tasks.Input;
+import qsl.internal.GroovyXml;
+import qsl.internal.json.ModJsonObject;
+
+public class QSLModuleExtension {
```

Comment:
I realize I'm a bit late and this is an extremely minor thing, but this class name does violate the Java class naming convention we usually use. Acronyms should be treated as words, so Qsl, not QSL